s3/s3_operations.cpp

```diff
@@ -751,9 +752,12 @@ namespace irods_s3 {
             rodsLog(developer_messages_log_level, "%s:%d (%s) [[%lu]] oprType set to %d\n",
                     __FILE__, __LINE__, __FUNCTION__, thread_id, oprType);
 
-            // fix open mode
             ios_base::openmode open_mode;
             if (oprType == PUT_OPR) {
+                // Fix open mode so we can stream
+                // The open mode is forced to O_CREAT, O_WRONLY, O_TRUNC
+                // If this follows with a read, then we are doing a read after write for checksum, the open_mode
+                // will be set back to the original value.
```

Comment:
I'm a little confused by the last two lines of the comment. Can you provide more explanation here in the comments?